code/script.js

```diff
@@ -254,90 +255,119 @@ const start = () => {
-      )
+      alert(`Yes, the person wears ${currentQuestion.attribute}! Keep all that wears ${currentQuestion.attribute}`)
     } else {
-      alert(
-        `No, the person doesn't wear ${attribute}! Remove all that wears ${attribute}`
-      )
+      alert(`No, the person doesn't wear ${currentQuestion.attribute}! Remove all that wears ${currentQuestion.attribute}`)
     }
-  } else if (group === 'other') {
-    // Similar to the one above
-  } else {
+  } else if (currentQuestion.category === 'eye color') {
+    if (keep) {
+      alert(`Yes, the person has ${currentQuestion.value} eyes! Keep all that has ${currentQuestion.value} eyes`)
+    } else {
+      alert(`No, the person doesn't have ${currentQuestion.value} eyes! Remove all that has ${currentQuestion.value} eyes`)
+    }  
+  } else if (currentQuestion.category === 'hair color') {
+    if (keep) {
+      alert(`Yes, the person has ${currentQuestion.value} hair! Keep all that has ${currentQuestion.value} hair`)
+    } else {
+      alert(`No, the person doesn't have ${currentQuestion.value} hair! Remove all that has ${currentQuestion.value} hair`)
+    }  
+  } else if (currentQuestion.category === 'other') {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all persons with yellow hair"
+      alert(`Yes the person smokes! Keep all the smokers`)
     } else {
-      // alert popup that says something like: "NO, the person doesnt have yellow hair! Remove all persons with yellow hair"
+      alert(`No the person doesn't smoke! Remove all the smokers`)
     }
   }
 
-  // filter to keep or remove based on the keep variable.
-  /* charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-    or 
-    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value) */
-
+  if (keep) {
+    charactersInPlay = charactersInPlay.filter((person) => person[currentQuestion.attribute] === currentQuestion.value)
+  } else {
+    charactersInPlay = charactersInPlay.filter((person) => person[currentQuestion.attribute] !== currentQuestion.value)
+  }
+  // filter to keep or remove based on the keep variable.    
   // Invoke a function to redraw the board with the remaining people.
+  generateBoard()
 }
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (suspect) => {
+  const confirmation = confirm(`Are you sure about your bet on ${suspect}?`)
+  if (confirmation) {
+    checkMyGuess(suspect)
+  } 
   // store the interaction from the player in a variable.
   // remember the confirm() ?
   // If the player wants to guess, invoke the checkMyGuess function.
 }
 
 // If you confirm, this function is invoked
 const checkMyGuess = (suspect) => {
+  if (suspect === secret.name) {
+    alert('That is correct, you win!')
+    board.innerHTML = ''
+    winOrLose.style.display = 'block'
```

Comment:
So this triggers an alert, but I cannot see the winorloose section. I think perhaps that you should exclude the alert and instead place your message inside the win or loose section by using .innerHTML.